src/link/MachO.zig

```diff
@@ -4374,7 +4375,7 @@ fn populateMissingMetadata(self: *MachO) !void {
             .segment = .{
                 .inner = .{
                     .segname = makeStaticString("__PAGEZERO"),
-                    .vmsize = pagezero_vmsize,
+                    .vmsize = self.base.options.pagezero_size orelse pagezero_vmsize,
```

Comment:
AFAIK you cannot just set the value to whatever you want without aligning to the page size.